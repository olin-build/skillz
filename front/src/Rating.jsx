/* eslint react/no-array-index-key: 0 */
import React, { Component } from 'react';

export const Rating = ({ rating, icon }) => {
  const size = 'tiny';
  return (
    <div className="compact my-rating">
      {Array(...Array(rating || 0)).map((_, i) =>
        <i key={i} className={`${size} ${icon || 'star'} icon`} />)}
      {!!rating || <i className={`${size} empty star icon`} style={{ color: 'transparent' }} />}
    </div>);
};

export class EditRating extends Component {
  constructor(props) {
    super(props);
    this.state = { rating: props.rating };
    this.onRating = this.onRating.bind(this);
  }
  onRating(rating) {
    this.props.onRating(rating);
    this.setState({ rating });
  }
  render() {
    const { props, state, onRating } = this;
    const icon = props.icon || 'star';
    const rating = state.rating || 0;
    const size = '';
    return (
      <div className="my-rating">
        <i
          className={`${size} circle icon${rating ? ' thin' : ''}`}
          onClick={() => onRating(null)}
        />
        {rating > 0 && Array(...Array(rating)).map((_, i) => (
          <i
            key={i}
            className={`${size} ${icon} icon`}
            onClick={() => onRating(i + 1)}
          />))}
        {rating < 5 && Array(...Array(5 - rating)).map((_, i) => (
          <i
            key={i + rating}
            className={`${size} empty ${icon} icon`}
            onClick={() => onRating(i + 1 + rating)}
          />))}
      </div>);
  }
}
