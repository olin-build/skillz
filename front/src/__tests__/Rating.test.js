import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { EditRating, Rating } from '../Rating';

describe('Rating', () => {
  test('matches snapshots', () => {
    [null, 0, 1, 5].forEach((rating) => {
      const component = renderer.create(<Rating rating={rating} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot(`rating=${rating}`);
    });
  });
  test('renders {rating} `.icon.star`', () => {
    // This is functionally redundant to the snapshot tests, but
    // it communicates the intent.
    const rating = shallow(<Rating rating={2} />);
    expect(rating.find('.icon.star')).toHaveLength(2);
  });
});


describe('EditRating', () => {
  test('matches snapshots', () => {
    [null, 0, 1, 5].forEach((rating) => {
      const component = renderer.create(<EditRating rating={rating} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot(`rating=${rating}`);
    });
  });
});
