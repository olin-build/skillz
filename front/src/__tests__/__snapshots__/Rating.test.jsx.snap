// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRating matches snapshots: rating=0 1`] = `
<div
  className="my-rating"
>
  <i
    className=" circle icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
</div>
`;

exports[`EditRating matches snapshots: rating=1 1`] = `
<div
  className="my-rating"
>
  <i
    className=" circle icon thin"
    onClick={[Function]}
  />
  <i
    className=" star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
</div>
`;

exports[`EditRating matches snapshots: rating=5 1`] = `
<div
  className="my-rating"
>
  <i
    className=" circle icon thin"
    onClick={[Function]}
  />
  <i
    className=" star icon"
    onClick={[Function]}
  />
  <i
    className=" star icon"
    onClick={[Function]}
  />
  <i
    className=" star icon"
    onClick={[Function]}
  />
  <i
    className=" star icon"
    onClick={[Function]}
  />
  <i
    className=" star icon"
    onClick={[Function]}
  />
</div>
`;

exports[`EditRating matches snapshots: rating=null 1`] = `
<div
  className="my-rating"
>
  <i
    className=" circle icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
  <i
    className=" empty star icon"
    onClick={[Function]}
  />
</div>
`;

exports[`Rating matches snapshots: rating=0 1`] = `
<div
  className="compact my-rating"
>
  <i
    className="tiny empty star icon"
    style={
      Object {
        "color": "transparent",
      }
    }
  />
</div>
`;

exports[`Rating matches snapshots: rating=1 1`] = `
<div
  className="compact my-rating"
>
  <i
    className="tiny star icon"
  />
</div>
`;

exports[`Rating matches snapshots: rating=5 1`] = `
<div
  className="compact my-rating"
>
  <i
    className="tiny star icon"
  />
  <i
    className="tiny star icon"
  />
  <i
    className="tiny star icon"
  />
  <i
    className="tiny star icon"
  />
  <i
    className="tiny star icon"
  />
</div>
`;

exports[`Rating matches snapshots: rating=null 1`] = `
<div
  className="compact my-rating"
>
  <i
    className="tiny empty star icon"
    style={
      Object {
        "color": "transparent",
      }
    }
  />
</div>
`;
